syft/internal/cache/dir.go

```diff
@@ -0,0 +1,71 @@
+package cache
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/adrg/xdg"
+	"github.com/mitchellh/go-homedir"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+// SetRootDir sets the root directory to use for downloaded content caching for things like module license information
+func SetRootDir(dir string) {
+	rootDir = dir
+}
+
+// rootDir if set, will be the default root directory used to cache content
+var rootDir = ""
+
+// findRoot returns a directory to be used for caching, first available in this order:
+//   - rootDir
+//   - XDG_CACHE_HOME
+//   - HOME/.cache
+//   - TEMP
+func findRoot() string {
+	if rootDir != "" {
+		dir, err := homedir.Expand(rootDir)
+		if err != nil {
+			log.Warnf("unable to expand cache directory %s: %v", rootDir, err)
+		} else {
+			return dir
+		}
```

Comment:
I feel like this should be in `SetRootDir` and done once. `homedir.Expand()` can be expensive, so invoking it as few times as possible would be ideal.

Reply:
actually, I might be wrong about this -- I think `homedir` caches it's result globally.

Reply:
This is only done "once", it's part of the configuration now.